src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs

```diff
@@ -203,6 +203,30 @@ private static void BeforeExecuteCommand(TypeCharCommandArgs args, CommandExecut
             }
         }
 
+        private static bool CanHaveSemicolon(SyntaxNode currentNode)
+        {
+            if (currentNode.IsKind(SyntaxKind.FieldDeclaration, SyntaxKind.DelegateDeclaration, SyntaxKind.ArrowExpressionClause))
+            {
+                return true;
+            }
+
+            if (currentNode is RecordDeclarationSyntax { OpenBraceToken: { IsMissing: true } })
+            {
+                return true;
+            }
+
+            if (currentNode is MethodDeclarationSyntax method)
+            {
+                if (method.Modifiers.Any(SyntaxKind.AbstractKeyword) || method.Modifiers.Any(SyntaxKind.ExternKeyword) ||
+                    method.IsParentKind(SyntaxKind.InterfaceDeclaration))
```

Comment:
❔ Is there any harm in making this unconditional?

Reply:
📝 Note that `partial` methods also can be declared without a body.

Reply:
@sharwell I think it will unnecessarily move the caret to the end for typos, which can be annoying.

Reply:
> 📝 Note that `partial` methods also can be declared without a body.

That's a good case. Going to fix it and add a test.